c/stcbackground.c

```diff
@@ -0,0 +1,98 @@
+#include "logging.h"
+#include "stcbackground.h"
+
+#define STC_BG_ENTRIES 101
+#define STC_BG_INTERVAL 10
+
+
+static int processInterval(STCBase *stcBase, STCModule *module, int selectStatus) {
+  STCCallbackList* moduleData = (STCCallbackList*)(module->data);
+  STCIntervalCallbackData* callbackList = moduleData->callbackList;
+
+  for (int i = 0; i < STC_BG_ENTRIES; i++) {
+    STCIntervalCallbackData* t = &callbackList[i];
+    if (t->id != -1)  /* initialise */ {
+      
+      // giving ability to disable job by making timeInterval less than zero
+      // as we return callbackdata, user can make timeInterval zero
+      if(t->timeInterval<0) continue;
+
+      t->countInterval+=STC_BG_INTERVAL;
+      if (t->countInterval <= 0) {
+        STCIntervalCallback callback = t->callback;
+        callback(moduleData->server, stcBase, module, t, t->userData);
+        t->countInterval = -t->timeInterval;
+      }
+    } else {
+      break;
+    }
+  }
+  return 0;
+};
+
+static int nextSlot(STCIntervalCallbackData* callbackList) {
+  for (int i = 0; i < STC_BG_ENTRIES; i++) {
+    if (callbackList[i].id == -1)  /* initialise */ {
+      return i;
+    }
+  }
+  return -1;
+};
+
+STCModule* stcInitBackgroundModule(void* server, STCBase *stcBase) {
+  STCCallbackList* moduleData = (STCCallbackList*)safeMalloc(sizeof(STCCallbackList),"STCCallbackList");
+  if (moduleData == NULL) {
+    return NULL;
+  }
+
+  STCIntervalCallbackData* callbackList = (STCIntervalCallbackData*)safeMalloc(sizeof(STCIntervalCallbackData)*STC_BG_ENTRIES,"STCIntervalCallbackData");
+  if (callbackList == NULL) {
```

Comment:
You're leaking `moduleData`. We'd also want to free both `callbackList` and `moduleData` if `stcRegisterModule` returns NULL.

Reply:
yes, ok

Reply:
doing allocation together now, so should be good now